torch/_subclasses/fake_tensor.py

```diff
@@ -621,7 +621,7 @@ def __init__(
         *,
         allow_fallback_kernels=True,
         allow_meta=False,
-        throw_on_data_dependent_ops=False,
+        throw_on_data_dependent_ops=True,
```

Comment:
I'm probably gonna dig more to confirm my understanding of the current state, but just to summarize - will dynamo catch "data dependent op" exceptions and fall back the entire current graph to eager? Or will it (smartly) graph break and try to compile the graph before + the graph after.

Reply:
Yes, graph break: https://github.com/pytorch/torchdynamo/blob/main/torchdynamo/variables/tensor.py#L152